js/ui/scheduler/ui.scheduler.subscribes.js

```diff
@@ -418,14 +418,15 @@ const subscribes = {
     },
 
     updateAppointmentEndDate: function(options) {
-        let endDate = new Date(options.endDate),
-            endDayHour = this._getCurrentViewOption('endDayHour'),
-            startDayHour = this._getCurrentViewOption('startDayHour'),
-            updatedEndDate = endDate;
+        const endDate = options.endDate;
+        const endDayHour = this._getCurrentViewOption('endDayHour');
+        const startDayHour = this._getCurrentViewOption('startDayHour');
+
+        let updatedEndDate = endDate;
 
         if(endDate.getHours() >= endDayHour) {
             updatedEndDate.setHours(endDayHour, 0, 0, 0);
-        } else if(startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
+        } else if(!options.sameDate && startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
```

Comment:
This need to be discussed later.

Reply:
updateAppointmentEndDate method is needed to check whether a **long** appointment is placed in the interval [startDayHour, endDayHour]. If not, we crop the **appointment part** on the previous day. This does not need to be done for **short appointments** that do not have a part on the previous day.

Reply:
Yes, you're absolutely right. But I think this logic should be renovated after we will implement 'FREQ=HOURLY' in our recurrence editor. What do you think about that, @Yarovaya?